src/features/cart/ShoppingCartTest.ts

```diff
@@ -4,171 +4,176 @@ interface Item{
     title:string,
     unitPrice:number,
     quantity:number,
-    discount:boolean
+    discount:string[]
 };
 
 
 interface DiscountCode{
     code:string,
+    type: "cart" | "item",
     value:number,
-    applied:string
+    itemId?: number;
 
 };
+interface Cart{
+    items:Item[],
+    discount:string[]
+}
 
-const discountCode:Array<DiscountCode>=[
+const discountCodes:Array<DiscountCode>=[
     {
         code:'NewYear2022',
+        type:"cart",
         value:20,
-        applied:'all'
     },
     {
         code:'summer2022',
+        type:"cart",
         value:10,
-        applied:'Mens-Plo T-shirt'
     },
     {
         code:'fullFree',
-        value:100,
-        applied:"Womens-Plo T-shirt"
+        type:"item",
+        value:70,
+        itemId:2
+        
     }
 
 ]
-const item:Item={
-    id:1,
-    title:'Mens TShirt',
-    unitPrice:35.99,
-    quantity:1,
-    discount:false
+const cart:Cart={
+    items:[],
+    discount:[]
 }
 
-let storeItem:Array<Item>=[];
-
 //Add to Product Card 
 const addToCart=(item:Item):void=>{
-    const product=storeItem.find((product)=>product.id==item.id);
-    if(product){
+    const isItem=cart.items.find((product)=>product.id===item.id)
+    if(isItem){
         item.quantity++
```

Comment:
I think it should be isItem.quantity++ as parameter item does not belong to cart.items array.

Reply:
Thanks I am Solve this